openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TestCaseRepository.java

```diff
@@ -927,4 +962,25 @@ public void entitySpecificUpdate() {
       recordChange("testCaseResult", original.getTestCaseResult(), updated.getTestCaseResult());
     }
   }
+
+  public TableData getSampleData(UUID testCaseID, boolean authorizePII) {
```

Comment:
we can add fields options as well for 'failedRowsSample' field

Reply:
there is no way to process the `authorizePII` field in the `setField` method. should I add it as another method in the `TestCaseResource`:

```java
testCase = getInternal(..)
boolean authorizePII = authorizer.authorizePII(securityContext, resourceContext.getOwner());
testCase.setFailedRowsSample(repository.getSampleData(testCase, authorizePII));
return testCase
```